### PR TITLE
[WIP] Rubocop update

### PR DESCRIPTION
**Pull request by lslezak**
I'm trying to update Rubocop to the latest version, let's see how much work that will require...

### CODE REVIEW

**Comment by jreidinger** [comment]
in general it looks good. We just need to be sure that there is no bugs for our code if we want to unify whole yast code on new rubocop version. ( BTW I use new version in some projects and there it works fine for me ( cfa e.g. )

File: src/lib/migration/proposal_client.rb [hunk 1]
```diff
@@ -173,7 +173,7 @@ def proposal_warning(warnings, links)
     def disable_repo(link)
       log.info "Activated link: #{link}"
 
-      link.match(/^#{LINK_PREFIX}(\d+)/)
+      link =~ /^#{LINK_PREFIX}(\d+)/
       repo = Regexp.last_match(1).to_i
```
Comment (jreidinger) [inline src/lib/migration/proposal_client.rb]: btw this can be written as `repo = link[/^#{LINK_PREFIX}(\d+)/, 1].to_i`

**Comment by jreidinger** [comment]
just NP, LGTM

File: src/lib/migration/proposal_client.rb (hunk 1 above)
Comment (lslezak) [inline src/lib/migration/proposal_client.rb]: Oh, a nice trick but less readable IMHO... I'll keep the current code.

**Pull request merged**